pkg/build/api/types.go

```diff
@@ -68,10 +75,71 @@ type BuildSpec struct {
 
 	// Optional duration in seconds, counted from the time when a build pod gets
 	// scheduled in the system, that the build may be active on a node before the
-	// system actively tries to terminate the build; value must be positive integer
+	// system actively tries to terminate the build; value must be positive
+	// integer
 	CompletionDeadlineSeconds *int64
+
+	// Info is a placeholder that can store multiple reasons why a build was
+	// triggered. It should cover for a future implementation of level triggered
+	// builds where you can multiple reasons why a build was triggered.
+	Info []*BuildTriggerInfo
+}
+
+// BuildTriggerInfo contains information about why/who/what triggered a build.
+type BuildTriggerInfo struct {
+	// Details is a placeholder for  information about who triggered a build if
+	// it's either a user, a token or something else
+	Details BuildTriggerDetails
+
+	// JenkinsJobsDetails holds details about the jenkins job that triggered this
+	// specific build
+	JenkinsJob JenkinsJobDetails
```

Comment:
@rhcarvalho me and @PI-Victor discussed this and we will probably opt-out jenkins support here for now. this should be a follow up card that will also include updating the Jenkins Plugin to set these informations.